python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -176,16 +179,22 @@ def _dispatch_tasks(self):
         # no-op in the streaming execution case.
         if self._bundle_queue:
             # Try to scale up if work remains in the work queue.
-            self._scale_up_if_needed()
+            self._scale_up_if_needed(len(self._bundle_queue))
         else:
             # Only try to scale down if the work queue has been fully consumed.
             self._scale_down_if_needed()
 
-    def _scale_up_if_needed(self):
-        """Try to scale up the pool if the autoscaling policy allows it."""
-        while self._autoscaling_policy.should_scale_up(
+    def _scale_up_if_needed(self, num_inputs: int):
+        """Try to scale up the pool if the autoscaling policy allows it.
+
+        Args:
+            num_inputs: The number of inputs for this operator.
+
+        """
+        if self._autoscaling_policy.should_scale_up(
```

Comment:
also need to change this from while -> if @ericl @c21 

I think this was a bug before...`_execution_allowed` only takes into account the incremental resource usage from scaling up a single actor.

Using a `while` autoscales too aggressively and may exceed the resource limits.